src/priorityq.py

```diff
@@ -9,18 +9,32 @@ def __init__(self):
         """Priority Queue constructor method."""
         self.dict = {}
 
-    def insert(self, key, val):
+    def insert(self, value, priority):
         """Priority Queue insert method."""
-        self.dict.setdefault(key, []).append(val)
+        if not isinstance(priority, int):
+            raise TypeError("Priority must be an integer")
+        if priority in self.dict:
+            self.dict[priority].append(value)
+        else:
+            self.dict[priority] = [value]
+        print(self.dict)
 
     def pop(self):
         """Priority Queue pop method."""
-        mostest_importantestest = self.dict[min(self.dict.keys())][0]
-        if min(self.dict.keys()) < 0:
-            self.dict[min(self.dict.keys())].pop
-        return mostest_importantestest
+        if len(self.dict):
+            cursor = 0
+            while cursor not in self.dict:
+                cursor += 1
+            next_node = self.dict[cursor][0]
+            self.dict[cursor] = self.dict[cursor][1:]
+            return next_node
+        else:
+            raise IndexError("The Priority Queue is empty")
 
     def peek(self):
         """Priority Queue peek method."""
-        mostest_importantestest = self.dict[min(self.dict.keys())][0]
-        return mostest_importantestest
+        if len(self.dict):
+            mostest_importantestest = self.dict[min(self.dict.keys())][0]
+            return mostest_importantestest
```

Comment:
Peek and pop will return different values if I insert something with a priority of 1 and another thing with a priority of -1.  That's going to be problematic because nothing in your code prevents me from doing that.  Add a test and fix the problem